projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.spec.ts

```diff
@@ -166,14 +201,74 @@ describe('LinkValueComponent', () => {
         }
       );
 
-      testHostComponent.inputValueComponent.searchByLabel('thing');
-      testHostFixture.detectChanges();
+      // simulate user searching for label 'thing'
+      testHostComponent.inputValueComponent.valueFormControl.setValue('thing');
```

Comment:
the point of this test was to check only the functionality of the searchByLabel directly, not through reading input value and passing it to the function. Furthermore, assigning a string 'thing' as you have done here is invalid.

Reply:
No, it's not in valid. Because this triggers `searchByLabel`.

Reply:
I had triggered the `searchByLabel` directly, what you do is triggering it while making the form invalid.